test/e2e/usersignup_test.go

```diff
@@ -602,6 +613,43 @@ func (s *userSignupIntegrationTest) TestCapacityManagementWithManualApproval() {
 
 		assert.Equal(t, toolchainv1alpha1.UserSignupStateLabelValueApproved, user.UserSignup.Labels[toolchainv1alpha1.UserSignupStateLabelKey])
 	})
+
+	s.T().Run("last provisioned space flips SpaceProvisionerConfig to not ready", func(t *testing.T) {
+		// given
+		// there are 4 already deployed spaces before this test is run, so we set the capacity to 5 so that 1 last space can be deployed.
```

Comment:
I don't think that this will really work as the Spaces were deployed in other sub-tests and the users should be cleaned automatically at the end of each of the sub-tests.
This may also lead to flakiness.
I would either transform it to a separate test:
```go
func (s *userSignupIntegrationTest) TestSpaceProvisionerConfigFlip() {
	hostAwait := s.Host()
	memberAwait1 := s.Member1()
	memberAwait2 := s.Member2()
	// given
	s.T().Run("last provisioned space flips SpaceProvisionerConfig to not ready", func(t *testing.T) {
		// given
		spaceprovisionerconfig.UpdateForCluster(t, hostAwait.Awaitility, memberAwait1.ClusterName, testSpc.MaxNumberOfSpaces(1))
		spaceprovisionerconfig.UpdateForCluster(t, hostAwait.Awaitility, memberAwait2.ClusterName, testSpc.Enabled(false))
```
or you can reuse an already existing test that does something similar
eg: https://github.com/codeready-toolchain/toolchain-e2e/blob/59268cf3699545643a490c841ea18fcdd05119c8/test/e2e/usersignup_test.go#L241-L282
where after updating the SPC you can also check the readiness:
```go
spaceprovisionerconfig.UpdateForCluster(t, hostAwait.Awaitility, memberAwait1.ClusterName, testSpc.MaxNumberOfSpaces(1))
spaceprovisionerconfig.UpdateForCluster(t, hostAwait.Awaitility, memberAwait2.ClusterName, testSpc.MaxNumberOfSpaces(1))
_, err := wait.For(t, hostAwait.Awaitility, &toolchainv1alpha1.SpaceProvisionerConfig{}).FirstThat(
	assertions.Has(spaceprovisionerconfig.ReferenceToToolchainCluster(memberAwait1.ClusterName)),
	assertions.Is(testSpc.Ready()))
require.NoError(t, err)
_, err = wait.For(t, hostAwait.Awaitility, &toolchainv1alpha1.SpaceProvisionerConfig{}).FirstThat(
	assertions.Has(spaceprovisionerconfig.ReferenceToToolchainCluster(memberAwait2.ClusterName)),
	assertions.Is(testSpc.Ready()))
require.NoError(t, err)
```
and then expect the flip to an unready state after provisioning the users

Reply:
Extracted this into a separate top-level test in the testsuite.